jni/src/Foreign/JNI.hs

```diff
@@ -435,11 +459,14 @@ setStatic/**/name/**/Field klass field x = \
     throwIfException env $ \
     [CU.block| void { \
       (*$(JNIEnv *env))->SetStatic/**/name/**/Field($(JNIEnv *env), \
-                                                    $(jclass klass), \
+                                                    $fptr-ptr:(jclass klass), \
                                                     $(jfieldID field), \
                                                     $(c_fieldtype x)); } |]
 
-SET_STATIC_FIELD(Object, JObject, jobject)
+setStaticObjectField :: JClass -> JFieldID -> JObject -> IO ()
+setStaticObjectField x y z =
+    let SET_STATIC_FIELD(Object, (Ptr JObject), jobject)
+    in withForeignPtr (coerce z) (setStaticObjectField x y)
```

Comment:
These definitions would read better like this:
```Haskell
setStaticObjectField :: JClass -> JFieldID -> JObject -> IO ()
setStaticObjectField x y z =
    let setStaticObjectPtrField = SET_STATIC_FIELD_RHS(Object, (Ptr JObject), jobject)
    in withForeignPtr (coerce z) (setStaticObjectPtrField x y)
```
It only would take factoring out SET_STATIC_FIELD so it uses `SET_STATIC_FIELD_RHS` as well.